capplets/time-admin/src/time-share.c

```diff
@@ -0,0 +1,154 @@
+            break;
+        }
+        case QUESTION:
+        {
+            dialog = gtk_message_dialog_new(GTK_WINDOW(WindowLogin),
+                                            GTK_DIALOG_DESTROY_WITH_PARENT,
+                                            GTK_MESSAGE_QUESTION,
+                                            GTK_BUTTONS_YES_NO,
+                                            "%s",Title);
+            gtk_dialog_add_button (GTK_DIALOG (dialog),("_Return"),
+                                   GTK_RESPONSE_ACCEPT);
+            break;
+        }
+        case QUESTIONNORMAL:
+        {
+            dialog = gtk_message_dialog_new(GTK_WINDOW(WindowLogin),
+                                            GTK_DIALOG_DESTROY_WITH_PARENT,
+                                            GTK_MESSAGE_QUESTION,
+                                            GTK_BUTTONS_YES_NO,
+                                            "%s",Title);
+            break;
+        }    
+        default :
+            break;
+
+    }
+    gtk_message_dialog_format_secondary_markup(GTK_MESSAGE_DIALOG(dialog),
+                                               TYPEMSG,
+                                               Msg);
+    gtk_window_set_title(GTK_WINDOW(dialog),("Message"));
+    nRet =  gtk_dialog_run(GTK_DIALOG(dialog));
+    gtk_widget_destroy(dialog);
+
+    return nRet;
+}
+void SetLableFontType(GtkWidget *Lable ,const char *Color,int FontSzie,const char *Word)
+{
+    char LableTypeBuf[200] = { 0 };
+
+    sprintf(LableTypeBuf,
+           "<span foreground=\'%s\'weight=\'light\'font_desc=\'%d\'>%s</span>",
+            Color,FontSzie,Word);
+    gtk_label_set_markup(GTK_LABEL(Lable),LableTypeBuf);
+
+}
+void QuitApp(TimeAdmin *ta)
+{
+    if(ta->UpdateTimeId > 0)
+    {
+        g_source_remove (ta->UpdateTimeId);
+    }    
+    if(ta->ApplyId > 0)
+    {
+        g_source_remove(ta->ApplyId);
+    }    
+    
+    gtk_main_quit();
+}
+GtkWidget* DialogAddButtonWithIconName (GtkDialog   *dialog,
+                                        const gchar *button_text,
+                                        const gchar *icon_name,
+                                        gint         response_id)
+{
+    GtkWidget *button;
+
+    button = gtk_button_new_with_mnemonic (button_text);
+    gtk_button_set_image (GTK_BUTTON (button), gtk_image_new_from_icon_name (icon_name, GTK_ICON_SIZE_BUTTON));
+
+    gtk_button_set_use_underline (GTK_BUTTON (button), TRUE);
+    gtk_style_context_add_class (gtk_widget_get_style_context (button), "text-button");
+    gtk_widget_set_can_default (button, TRUE);
+//    gtk_widget_show (button);
+    gtk_dialog_add_action_widget (GTK_DIALOG (dialog), button, response_id);
+
+    return button;
+}
+void SetTooltip(GtkWidget*box,gboolean mode)
+{
+    if(!mode)
+    {
+        gtk_widget_set_tooltip_markup(box,
+                                     _("Network time synchronization has been set up.prohibit Modification Time"));
```

Comment:
Grammar mistake:
Network time synchronization has been set up. Manual time/date setting is disabled.